### ISSUE TITLE
Update documentation to follow the documentation style guide

### ISSUE THREAD

**Issue by asmeurer** [sympy/sympy#19591]
This issue should serve as a master issue for tracking which parts of the documentation follow the style guide that was developed as part of GSoD 2019 https://docs.sympy.org/latest/documentation-style-guide.html.

So far, following modules have been updated

- solvers (https://github.com/sympy/sympy/pull/17921)
- functions.special (https://github.com/sympy/sympy/pull/17844)

**Comment by Soumi7** [comment]
Will start working on implementing the style guide to functions.elementary.

**Comment by Soumi7** [comment]
@Sc0rpi0n101 
I will implement the style guide for elementary submodule in functions, you suggested having the Parameters section before examples. 

Should I then add the Parameters list before the examples when I add them here?

**Comment by Sc0rpi0n101** [comment]
I think we should finish the discussion regarding the Docstring sections and update the Documentation style guide before starting with the implementation. 

Updating the docstrings now and then updating the guide later would require them to be changed later again. That would just be unnecessary extra work. 
It would be better to change the required section order, add the missing sections, etc. in the style guide and change the docstrings as per the updated style guide.

**Comment by Soumi7** [comment]
Yes I think that too.

**Comment by Soumi7** [comment]
![kk](https://user-images.githubusercontent.com/51290447/85166286-af052a80-b284-11ea-9c52-f31a80380eab.png)
This is my proposed flowchart of the docstring sections and their order. Suggestions are welcome!

**Comment by asmeurer** [comment]
These things were already discussed when we wrote the guide last year so we should stick with what the guide says now. If we want to change the guide we should start a new discussion, but it shouldn't block any work to update things to the current guide. Also any such discussion should reference the discussions from last year that lead to the current way things are.

**Comment by asmeurer** [comment]
Note that most of the discussions about this happened at https://github.com/sympy/sympy/pull/17715. I don't recall if we discuss this on that issue or in the private GSoD calls, but I do remember we explicitly chose to put examples up front.

**Comment by Soumi7** [comment]
As discussed before, I will stick to the latest guide for updating the docstrings. In #19613, I have added the same order in the docstring in the latest documentation style guide developed in GSoD 2019, but the rendering has produced slightly different results.

**Comment by asmeurer** [comment]
Regarding "returns", see also https://github.com/sympy/sympy/issues/17644

**Comment by Soumi7** [comment]
#19753

**Comment by asmeurer** [comment]
Adding a documentation page to the Sphinx docs is fairly straightfoward. If you look in the doc/ directory you can just mimic what is already there. However, it can sometimes be difficult because existing documentation can have build errors or formatting issues, which would have to be fixed.

**Comment by Soumi7** [comment]
Yes, will first make a list of all which are missing doc page by updating the comment, and then I will try to add the page to Sphinx docs.
